parser.go

```diff
@@ -260,8 +263,13 @@ func readSymbol(rd *Reader, init rune) (Expr, error) {
 	return Symbol(s), nil
 }
 
-func readKeyword(rd *Reader, init rune) (Expr, error) {
-	token, err := readToken(rd, init)
+func readKeyword(rd *Reader, _ rune) (Expr, error) {
+	r, err := rd.NextRune()
```

Comment:
This can be simpler. Keep it the way it was before and instead of passing `init` to `readToken` , simply pass `-1`.

```golang
token, err := readToken(rd, -1)
```

Reply:
Ah, cool!  I suspected there was a simpler way! 😃 

Fixed.